pkg/util/keys.go

```diff
@@ -58,9 +77,35 @@ func KeyForGroupVersionKind(group, version, kind string) string {
 	if group == "" {
 		return fmt.Sprintf("%s/%s", version, kind)
 	}
+
 	return fmt.Sprintf("%s/%s/%s", group, version, kind)
 }
 
+// Create a string key in the form group/version/resource or version/resource if the group is empty
+func KeyForGroupVersionResource(group, version, resource string) string {
```

Comment:
FYI: It is confusing that some methods named "KeyFor..." return a Key and others named the same way return a string.

Reply:
Such naming was already there, I'll open an issue for the rework i keep referring to, which should contain renamings to these functions if they remain.